porogo/api/models.py

```diff
@@ -1,3 +1,122 @@
+        email (str): Email of dev
+        short_description (str): Description of the dev
+        description_experience (str): Description of the dev's experience
+        description_personal (str): Description of the dev's personal interests
+        github_name (str): Github username of dev
+        linkedin_url (str): Linkedin url for the dev
+        other_url (str): Url to another site
+        other_url_type (str): Type of the other url
+        picture (File): Profile picture of the dev
+        featured_project (Project): Project that the user wants to display
+    """
+    first_name = models.CharField(max_length=20)
+    last_name = models.CharField(max_length=20)
+    email = models.EmailField()
+    short_description = models.TextField(max_length=512)
+    description_experience = models.TextField(max_length=1024)
+    description_personal = models.TextField(max_length=1024)
+    github_name = models.CharField(max_length=20, unique=True)
+    linkedin_url = models.URLField(max_length=200, null=True)
+    other_url = models.URLField(max_length=200, null=True)
+    other_url_type = models.CharField(max_length=20, null=True)
+    picture = models.ForeignKey(Image, on_delete=models.CASCADE)
+    featured_project = models.ForeignKey("Project", null=True, on_delete=models.CASCADE)
+    models.UniqueConstraint(fields=["first_name", "last_name"], name="unique_full_name")
+
+class Tag(models.Model):
+    """
+    Stores Tags for many to many relationships
+
+    Attributes:
+        tag (str): name of the tag
+    """
+    tag = models.CharField(max_length=16, unique=True)
+
+class Project(models.Model):
+    """
+    Stores Projects on the site
+
+    Attributes:
+        name (str): Name of the project
+        short_description (str): Short description of the project
+        description_intro (str): Introduction of the project
+        description_body (str): Description body of the project
+        icon (File): Icon of the project
+        github_link (str): github url for the repo
+        youtube_url (str): Youtube Url for showcase video
+        languages (Language[]): List of languages used in the project
+        tags (Tag[]): List of Tags for the project
+        contributors (Developer[]): List of contributors to the project
+        project_status (str): Current status of the project
+        start_year (int): Year the project was started
+        end_year (int): Year the project ended
+    """
+    PROJECT_STATUSES = {
+        "O": "Ongoing",
+        "U": "Upcoming",
+        "C": "Completed",
+        "H": "On Hold",
+    }
+
+    name = models.CharField(max_length=31, unique=True)
+    short_description = models.TextField(max_length=255)
+    description_intro = models.TextField(max_length=512)
+    description_body = models.TextField(max_length=1024)
+    icon = models.ForeignKey(Image, on_delete=models.CASCADE)
+    github_link = models.ForeignKey(URL, on_delete=models.CASCADE)
```

Comment:
Wouldn't there only be one github link for the project?

Reply:
I think we wanted to account for if a project had multiple repos? But even if that's the case that's not how to do it so I'll check what the intention was.

Reply:
We spoke about handling many links, updated accordingly